.github/workflows/feature-build-hniesb.yml

```diff
@@ -1,36 +1,154 @@
-  build:
+  # This job runs the maven commands to compile code and build package.
+  # This job also creates docker image and push the image to OpenShift repository
+  createHniEsbFeatureBuild:
     # The type of runner that the job will run on
     runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      OPENSHIFT_IMAGE_SHA: ${{ steps.push-image.outputs.digest }}
+      FEATUREBUILD_NAME: ${{ env.FEATUREBUILD_NAME }}
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v3
+      # It should get the code from branch that triggered this workflow
+      - name: Check out code
+        id: checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{github.event.inputs.BRANCH_NAME}}
 
-      # Runs a single command using the runners shell
-      - name: Run a one-line script
-        run: echo Hello, world!
 
-      # Runs a set of commands using the runners shell
-      - name: Run a multi-line script
+      # 1. Determine build name i.e. FeatureBuild-<branch-name>-<build execution count> 
+      - name: Determine Build name
+        run: |
+          echo "FEATUREBUILD_NAME=FeatureBuild-${{github.event.inputs.BRANCH_NAME}}-${{github.run_number}}" | tee -a $GITHUB_ENV
+        
+      # 2. Determine tags for OpenShift image
+      - name: Determine image tags
+        run: |
+          echo "IMAGE_TAGS= ${GITHUB_SHA::12} ${{env.FEATUREBUILD_NAME}}" | tee -a $GITHUB_ENV
+
+      # Run the maven install for common code
+      - name: Maven run for common code
         run: |
-          echo Add other actions to build,
-          echo test, and deploy your project.
+          cd ${{github.workspace}}/hn-common
+          mvn install
+
+      # Run the maven install for hnsesb code
+      - name: Maven run for hnsecure code
+        run: |
+          cd ${{github.workspace}}/hnsecure
+          mvn install -Dimpl-version=${{env.FEATUREBUILD_NAME}} -Dgit-commit=${GITHUB_SHA::12}
+
+      # Build from docker file
+      - name: Build hnsesb image from Dockerfile
+        id: build-docker-hniesb
+        uses: redhat-actions/buildah-build@v2
+        with:
+          image: ${{ env.IMAGE_NAME }}
+          tags: ${{ env.IMAGE_TAGS }}
+          context: ./hnsecure
+          dockerfiles: |
+            ./hnsecure/Dockerfile
+
+      # Upload the images in image repository in tools namespace
+      - name: Push image to registry
+        id: push-image
+        uses: redhat-actions/push-to-registry@v2
+        with:
+          image: ${{ steps.build-docker-hniesb.outputs.image }}
+          tags: ${{ steps.build-docker-hniesb.outputs.tags }}
+          registry: ${{env.IMAGE_REGISTRY}}/${{env.DEFAULT_NAMESPACE}}
+          username: ${{env.IMAGE_REGISTRY_USER}}
+          password: ${{env.OPENSHIFT_TOKEN}}
```

Comment:
Inconsistent spacing on braces compared to previous lines.

Reply:
updated